api/src/main/java/org/apache/iceberg/actions/ComputePartitionStats.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.PartitionStatisticsFile;
+
+/** An action that computes and writes the partition statistics of an Iceberg table. */
+public interface ComputePartitionStats
+    extends Action<ComputePartitionStats, ComputePartitionStats.Result> {
+  /**
+   * Choose the table snapshot to compute partition stats, by default the current snapshot is used.
```

Comment:
the `by default the current snapshot is used` part is rather confusing imo. The API requires the caller to pass the snapshot ID, so I don't see where the default would be applied here

Reply:
The design is similar to existing table stats (puffin). 

I followed the same java doc as table stats. https://github.com/apache/iceberg/blob/31c315f695aad544a096a5a2ffdde54a97b90b28/api/src/main/java/org/apache/iceberg/actions/ComputeTableStats.java#L39

Implementation use the `table.currentSnapshot` in the constructor. 

Do, you think we should remove it from the javadoc here?

Reply:
to me this rather feels like something that should be in the javadoc of the interface itself and not at the method level that actually takes a snapshotId as a parameter